force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,165 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler{
+
+    // Private Collection Variables
+
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id, Opportunity> newOppMap;
+    private Map<Id, Opportunity> oldOppMap;
+
+    // Constructor
+    public OpportunityTriggerHandler() {
+        this.newOppList = (List<Opportunity>) Trigger.new;
+        this.oldOppList = (List<Opportunity>) Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>) Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>) Trigger.oldMap;        
+    }
+
+    //* Opportunity Triggers
+
+    public override void beforeInsert() {
+        // Set default Type for new Opportunities
+        for (Opportunity opp : this.newOppList) {
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }
+        }
+    }
+
+    public override void afterInsert() {
+        //* Create a new Task for newly inserted Opportunities
+        List<Task> tasks = new List<Task>();
+        for (Opportunity opp : this.newOppList){
+            Task tsk = new Task();
+            tsk.Subject = 'Call Primary Contact';
+            tsk.WhatId = opp.Id;
+            tsk.WhoId = opp.Primary_Contact__c;
+            tsk.OwnerId = opp.OwnerId;
+            tsk.ActivityDate = Date.today().addDays(3);
+            tasks.add(tsk);
+        }
+        if (tasks.size() != 0) {
+            insert tasks;
+        }
+    }    
+    
+    public override void beforeUpdate() {
+        for(Opportunity opp : this.newOppList){
+            //* When an opportunity is updated validate that the amount is greater than 5000.    
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
```

Comment:
Try creating a constant string variable for the string value that goes into the error so it's easier to understand and change in the future.